src/code/z_vimode.c

```diff
@@ -1,158 +1,190 @@
-
-    cond_4C = not_70 && not_78;
-    cond_48 = not_70 && unk_78;
-    cond_44 = unk_70 && unk_78;
-    cond_40 = unk_70 && not_78;
-    cond_3C = unk_70 && not_74 && unk_78 && unk_7C;
-    cond_38 = unk_70 && unk_74 && unk_78 && not_7C;
-    cond_34 = not_70 && unk_74 && unk_78 && not_7C;
-
-    unk_top &= ~1;
-    unk_bottom &= ~1;
-
-    yScaleLo = (cond_4C ? 2 : 1) * ((height << 11) / (SCREEN_HEIGHT * 2 + unk_bottom - unk_top) / (unk_70 ? 1 : 2));
-
-    yScaleHi0 = not_78 ? (cond_40 ? 0x1000000 : 0x2000000) : 0;
-    yScaleHi1 = not_78 ? (cond_40 ? 0x3000000 : 0x2000000) : 0;
-
-    viMode->customViMode.type = mode;
-    viMode->customViMode.comRegs.ctrl = OS_VI_UNK2000 | OS_VI_UNK1000 | OS_VI_GAMMA | OS_VI_GAMMA_DITHER |
-                                        (!cond_44 ? OS_VI_UNK40 : 0) | (not_74 ? OS_VI_DIVOT : 0) |
-                                        (not_7C ? OS_VI_UNK2 | OS_VI_UNK1 : OS_VI_UNK2);
-
-    if (cond_3C) {
-        viMode->customViMode.comRegs.ctrl |= 0x100;
-    } else if (cond_38 | cond_34) {
-        viMode->customViMode.comRegs.ctrl |= 0x300;
-    } else if (unk_74) {
-        viMode->customViMode.comRegs.ctrl |= 0x200;
+    s32 yScaleHiEvenField;
+    s32 yScaleHiOddField;
+
+    hiRes = !loRes;
+    antialiasOn = !antialiasOff;
+    modeF = !modeN;
+    fb32Bit = !fb16Bit;
+
+    hiResDeflicker = hiRes && modeF;
+    hiResInterlaced = hiRes && modeN;
+    loResDeinterlaced = loRes && modeN;
+    loResInterlaced = loRes && modeF;
+
+    modeLAN1 = loRes && antialiasOn && modeN && fb16Bit;
+    modeLPN2 = loRes && antialiasOff && modeN && fb32Bit;
+    modeHPN2 = hiRes && antialiasOff && modeN && fb32Bit;
+
+    upperAdjust &= ~1;
+    lowerAdjust &= ~1;
+
+    yScaleLo =
+        (hiResDeflicker ? 2 : 1) * ((height << 11) / (SCREEN_HEIGHT * 2 + lowerAdjust - upperAdjust) / (loRes ? 1 : 2));
+
+    yScaleHiEvenField = modeF ? (loResInterlaced ? (F210(0.25) << 16) : (F210(0.5) << 16)) : 0;
+    yScaleHiOddField = modeF ? (loResInterlaced ? (F210(0.75) << 16) : (F210(0.5) << 16)) : 0;
+
+    viMode->customViMode.type = type;
+    viMode->customViMode.comRegs.ctrl = VI_CTRL_PIXEL_ADV(3) | VI_CTRL_GAMMA_ON | VI_CTRL_GAMMA_DITHER_ON |
+                                        (!loResDeinterlaced ? VI_CTRL_SERRATE_ON : 0) |
+                                        (antialiasOn ? VI_CTRL_DIVOT_ON : 0) |
+                                        (fb32Bit ? VI_CTRL_TYPE_32 : VI_CTRL_TYPE_16);
+
+    if (modeLAN1) {
+        // Anti-aliased, fetch extra lines as-needed
+        viMode->customViMode.comRegs.ctrl |= VI_CTRL_ANTIALIAS_MODE_1;
+    } else if (modeLPN2 | modeHPN2) {
+        // Point-sampled, resampling disabled
+        viMode->customViMode.comRegs.ctrl |= VI_CTRL_ANTIALIAS_MODE_3;
     } else {
-        viMode->customViMode.comRegs.ctrl |= 0;
+        if (antialiasOff) {
+            // Point-sampled, resampling enabled
+            viMode->customViMode.comRegs.ctrl |= VI_CTRL_ANTIALIAS_MODE_2;
```

Comment:
Any reason to not merge this into an `else if` like it was?

Reply:
The conditions being checked are of a different character (full mode vs only AA on/off), I think it's a bit clearer to see what's going on this way